app/controllers/discover.js

```diff
@@ -303,10 +303,11 @@ export default Ember.Controller.extend(Analytics, {
 
         const sort = {};
         const i18n = this.get('i18n');
+        const sortByOption = this.get('sortByOption').toString();
 
-        if (this.get('sortByOption').toString() === i18n.t('discover.sort_oldest_newest').toString()) {
+        if (sortByOption.toString() === i18n.t('discover.sort_oldest_newest').toString()) {
```

Comment:
You can get rid of the `.toString()` here and on L310, since you already have it as a string from L306.